src/core/include/openvino/core/except.hpp

```diff
@@ -16,15 +16,14 @@ namespace ov {
 /// Base error for ov runtime errors.
 class OPENVINO_API Exception : public std::runtime_error {
 public:
-    OPENVINO_DEPRECATED("This constructor is deprecated and will be removed, please use OPENVINO_THROW instead")
-    explicit Exception(const std::string& what_arg);
-
     [[noreturn]] static void create(const char* file, int line, const std::string& explanation);
     virtual ~Exception();
 
     static const std::string default_msg;
 
 protected:
+    explicit Exception(const std::string& what_arg);
```

Comment:
Check if `OPENVINO_SUPPRESS_DEPRECATED_START` macro can be removed in `AssertFailure` ctor.